src/main/host/host.rs

```diff
@@ -601,9 +601,11 @@ impl Host {
         res
     }
 
-    pub fn get_next_packet_priority(&self) -> f64 {
+    pub fn get_next_packet_priority(&self) -> u64 {
         let res = self.packet_priority_counter.get();
-        self.packet_priority_counter.set(res + 1.0);
+        // do a saturating add since if we did ever reach this number, we wouldn't want new packets
+        // to have a better priority
+        self.packet_priority_counter.set(res.saturating_add(1));
```

Comment:
What happens if we somehow reach this? Out of order packet delivery? Nondeterminism in the simulation?

I think `checked_add` with `unwrap` would be preferable. We really shouldn't be able to increment through all the u64s in practice.

Reply:
If we were to reach this, packets would still be in order and the simulation should still be deterministic (unless the fifo qdisc has a non-determinism bug). We already have packets that have duplicate packet priorities, such as control packets that all have a priority of 0.

I'll change this to a panic.